neurobooth_os/transfer_data.py

```diff
@@ -0,0 +1,66 @@
+"""
+    Moves data from local storage to network storage
+"""
+import os
+from subprocess import PIPE, Popen, STDOUT, CalledProcessError
+import argparse
+
+from neurobooth_os import config
+from neurobooth_os.util.constants import NODE_NAMES
+from neurobooth_os.log_manager import make_default_logger
+
+
+def log_output(logger, pipe):
+    for line in iter(pipe.readline, b''):  # b'\n'-separated lines
+        logger.info(str(line, "utf-8").strip('\r\n'))
+
+
+def main(args: argparse.Namespace):
+
+    logger = make_default_logger()
+
+    destination = config.neurobooth_config["remote_data_dir"]
+
+    source = config.neurobooth_config[args.source_node_name]["local_data_dir"]
+
+    try:
+        # Move data to remote
+        process = Popen(["robocopy", "/MOVE", source, destination, "/e"], stdout=PIPE, stderr=STDOUT)
+        with process.stdout:
+            log_output(logger, process.stdout)
+        return_code = process.wait()
+        logger.info(f"Transfer data to remote. Return code: {return_code}")
+
+        # Recreate local data folder
+        if not os.path.exists(source):
+            process = Popen(["mkdir", source], stdout=PIPE, stderr=STDOUT)
+            with process.stdout:
+                log_output(logger, process.stdout)
+            return_code = process.wait()
+            logger.info(f"Recreate local data directory. Return code: {return_code}")
+
+
+    except (OSError, CalledProcessError) as exception:
```

Comment:
Since this is the script-level error catch, would just catch the `Exception` class to make sure we don't miss something unexpected. An alternative would be to add an extra `except Exception as exception:` clause below with slightly different logging messages to reflect the unanticipated class.